src/EventLoop/Suspension.php

```diff
@@ -20,32 +18,26 @@
 final class Suspension
 {
     private ?\Fiber $fiber;
-    private \Fiber $scheduler;
-    private Driver $driver;
+
     private bool $pending = false;
 
+    /** @var callable */
+    private $interrupt;
+
     /**
-     * Suspension constructor.
-     *
      * @param Driver $driver
      * @param \Fiber $scheduler
+     * @param callable $interrupt
      *
      * @internal
      */
-    public function __construct(Driver $driver, \Fiber $scheduler)
-    {
-        $this->driver = $driver;
+    public function __construct(
+        private Driver $driver,
+        private \Fiber $scheduler,
```

Comment:
I'd like to avoid using these, especially on classes that also have other properties, as it makes it harder to understand which properties a class has.

Reply:
I've started using these frequently as it reduces a lot of boilerplate. I'll put it back though as it's not important here.